cypress/tests/organization-users.spec.ts

```diff
@@ -0,0 +1,420 @@
+    cy.wrap(false).as('testCompleted');
+
+    cy.wrap(4050).as('orgId');
+
+    // Locate the user row by username
+    cy.get('@orgUserTable')
+      .find('tr')
+      .filter(`:contains(${TEST_DATA.testUser.username})`)
+      .as('userRow');
+
+    // Click the user row to open details
+    cy.get('@userRow').click();
+
+    // Deactivate the user
+    cy.findByText(cyTranslate('deactivateUser')).click();
+    cy.findByText(cyTranslate('confirm')).click();
+
+    // Wait for the user edit action to complete
+    cy.wait('@editUser', NETWORK_WAIT_OPTS).then((interception) => {
+      expect(interception.response?.body?.is_active).to.eq(false);
+
+      expect(interception.response?.statusCode).to.eq(200);
+
+      // Store `userId` for cleanup
+      cy.wrap(interception.response?.body?.id).as('userId');
+    });
+
+    // Close the modal after deactivating the user
+    cy.findByTestId('member-drawer-close-btn').click();
+
+    // Enable viewing inactive members
+    cy.findByLabelText(cyTranslate('includeInactiveMembers')).check();
+
+    cy.wait('@membersList', NETWORK_WAIT_OPTS);
+
+    cy.findByTestId('org-user-table', ELEMENT_WAIT_OPTS).as('orgUserTable');
+
+    cy.wrap(() => {
+      cy.getAliases(['@userId', '@orgId']).then(([userId, orgId]) => {
+        // Make the API request to reactivate the user
+        cy.makeAuthenticatedAPIRequest({
+          method: 'PUT',
+          url: `https://api.appknox.com/api/organizations/${orgId}/users/${userId}`,
+          body: { is_active: true },
+        });
+      });
+    }).as('testCleanup');
+
+    // Locate the inactive user row by username
+    cy.get('@orgUserTable', ELEMENT_WAIT_OPTS)
+      .find('tr')
+      .filter(`:contains(${TEST_DATA.testUser.username})`)
+      .as('inactiveUserRow');
+
+    // Verify that the user is now inactive
+    cy.get('@inactiveUserRow', ELEMENT_WAIT_OPTS).within(() => {
+      cy.findByText(cyTranslate('chipStatus.inactive')).should('exist');
+    });
+
+    // Click the inactive user row to activate the user
+    cy.get('@inactiveUserRow').click();
+
+    // // Activate the user
+    cy.findByText(cyTranslate('activateUser')).click();
+    cy.findByText(cyTranslate('confirm')).click();
+
+    // Wait for the user edit action to complete
+    cy.wait('@editUser', NETWORK_WAIT_OPTS).then((interception) => {
+      expect(interception.response?.body?.is_active).to.eq(true);
+
+      expect(interception.response?.statusCode).to.eq(200);
+    });
+
+    // Close the modal after activation
+    cy.findByTestId('member-drawer-close-btn').click();
+
+    cy.wrap(true).as('testCompleted');
+  });
+
+  afterEach(function () {
+    if (shouldRunAfterEach) {
+      cy.get<boolean>('@testCompleted').then((testCompleted) => {
+        if (!testCompleted) {
+          cy.get('@testCleanup').then((testCleanup) => {
+            // Call the cleanup function
+            (testCleanup as unknown as () => void)();
+          });
```

Comment:
refactor clean up to `cy.wrap({ perform: () => { ...cleanup logic } })` and then you can directly invoke like this `cy.get('@testCleanup').invoke('perform')`